model/deepseek.go

```diff
@@ -0,0 +1,141 @@
+package model
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+
+	"github.com/sashabaranov/go-openai"
+)
+
+type DeepSeekProvider struct {
+	subType     string
+	apiKey      string
+	temperature float32
+	topP        float32
+}
+
+func NewDeepSeekProvider(subType string, apiKey string, temperature float32, topP float32) (*DeepSeekProvider, error) {
+	return &DeepSeekProvider{
+		subType:     subType,
+		apiKey:      apiKey,
+		temperature: temperature,
+		topP:        topP,
+	}, nil
+}
+
+func (p *DeepSeekProvider) GetPricing() string {
+	return `URL:
+https://api-docs.deepseek.com/zh-cn/quick_start/pricing
+
+| Model          | sub-type             | Input Price per 1K characters    | Output Price per 1K characters |
+|----------------|----------------------|----------------------------------|--------------------------------|
+|    deepseek    |     deepseek-chat    | 0.0001yuan/1,000 tokens          | 0.002yuan/1,000 tokens         |
```

Comment:
I think we should assume it's the cache miss case, so 0.001 instead of 0.0001

![image](https://github.com/user-attachments/assets/426fad1b-f395-47ee-bffa-8f20774e9cd1)